### ISSUE TITLE
White flash on screen push.

### ISSUE THREAD

**Issue by sexykodo** [wix/react-native-navigation#358]
### Issue Description
In IOS when i push a new screen i see a white background on the rootview before the first react native render, since i have a dark styled app it's quite annoying.

Changing the color on rootview in AppDelegate.m to black,
or setting a screenBackgroundColor doesn't seem to to anything 

### Steps to Reproduce / Code Snippets / Screenshots

This is the snippet from AppDeletage.m
`self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  self.window.backgroundColor = [UIColor blackColor];`

This is the snipper from the screenComponent

`class EventScreen extends Component {
    static navigatorStyle = {
        screenBackgroundColor: SituaColors.darkBg
    };
   ....
`
Thanks
---
### Environment
* React Native Navigation version: 2.0.0-experimental.105
* React Native version: 0.34.0
* Platform(s) (iOS, Android, or both?): IOS
* Device info (Simulator/Device? OS version? Debug/Release?): Both simulator and device, both debug and release

**Comment by alexcurtis** [comment]
Would be great to get some options for this. I also have a dark themed app and the white flashes between scenes are frustrating.

**Comment by DavoCg** [comment]
Do we know why this kind of flashing occurs ? 
I had not them before and since few changes in my code I had one flash only between two specific screens.

**Comment by ryanscottaudio** [comment]
same issue here; happens on push between two screens with `navBarHidden: true`

**Comment by savelichalex** [comment]
I add this as temp solution for my project under `setStyleOnAppearForViewController` in `RCCViewController`
```objective-c
NSString *screenColor = self.navigatorStyle[@"screenColor"];
    if (screenColor)
    {
      UIColor *color = screenColor != (id)[NSNull null] ? [RCTConvert UIColor:screenColor] : nil;
      viewController.view.backgroundColor = color;
    }
```

**Comment by sexykodo** [comment]
@savelichalex thanks! works like a charm!

**Comment by DavoCg** [comment]
Thanks @savelichalex it works well but does somebody a way to avoid this flashes ? maybe with preloading or idk ?

**Comment by savelichalex** [comment]
Maybe @guyca help with it, but I don't think so. As I understand how it works, when `push` happened VC create new native view(1), then they send event to js that render react view, and then rendered new react view added to native view(2). Between this two phases we see default native view with default background color.

**Comment by guyca** [comment]
Hey guys this issue will be addressed soon, I hope by the end of November. Sorry for the inconvenience. 

@savelichalex you're correct, that's exactly what happens. On Android we avoid this issue by showing the pushed screen only after `render` is done and the native view is populated.

**Comment by pascalmerme** [comment]
Hello, do you have an idea when pre-rendering will be available on iOS so we don't see anymore this white screen when pushing view or showing modal? Or maybe you can point us what should be changed so we can try to work on it? Thanks!

**Comment by brunolemos** [comment]
This is a big problem when switching tabs for the first time.
Thinking about migrating to a javascript-based tab navigator.

**Comment by ryanscottaudio** [comment]
@brunolemos yeah, I've been using NavigationExperimental for now

**Comment by DanielZlotin** [comment]
@brunolemos @ryanscottaudio  see #437

**Comment by grundmanise** [comment]
I am experiencing the same issue when using `switchToTab`. Tab's screen content flashes when it is switched for the first time. It is even worse for me because I am using a custom (JS) `tabbar` and it flashes too.

**Comment by jbolter** [comment]
@shahen94 This doesn't fix the flash issue for me on iOS.  Even if I add a 1 second delay it still seems that the view controller is not being rendered until the navigation transition actually starts.  I still see a brief flash of white (background color) as the transition happens.

**Comment by shahen94** [comment]
@jbolter it fixed my issue. seems like your JS thread is doing much work, Try to use `InteractionManager`

**Comment by guyca** [comment]
I don't think adding a delay before making the controller visible is recommended. On Android we solved this issue easily be display the screen only after the react root view has been populated. Meaning, once the react root view has at least one child - we conclude it's rendered and show it. Perhaps the same logic can be implemented on iOS as well.

**Comment by jbolter** [comment]
@shahen94 I'm doing very little work on the JS thread so I don't think that is the issue.  I can reproduce this in a very simple project if that helps debug.

@guyca Your proposed solution that works on Android sounds like a better solution than a delay.  Do you have any tips on where I can look to start investigating this solution?  I'm familiar with iOS development.

**Comment by guyca** [comment]
Unfortunately I'm not familiar with the iOS codebase. Based on the solution suggested here, I can only suggest you start looking at `RCCViewController` and make your way from there.

**Comment by DanielZlotin** [comment]
Guy I can find some time tomorrow to do that on iOS - point me to what you
meant in Android, I don't remember us doing anything like that..

**Comment by guyca** [comment]
Hey @DanielZlotin,
In [ContentView](https://github.com/wix/react-native-navigation/blob/master/android/app/src/main/java/com/reactnativenavigation/views/ContentView.java#L64), Once the first child is added to the `ReactRootView`, we call the `onDisplayListener` in `runOnPreDraw`

**Comment by jbolter** [comment]
@DanielZlotin Just a reminder that I would love for you to take a look at this "flashing" issue when yu have time!

**Comment by jbolter** [comment]
@DanielZlotin Would you have time to point me in the right direction? I am familiar with iOS so maybe I could take a crack at this.

**Comment by grundmanise** [comment]
Maybe it is possible to postpone route change until Components are rendered. But it sounds more of a 'hack'..

**Comment by jbolter** [comment]
Yeah it's unclear to me why the components aren't being rendered until the view is already appearing.  It seems like the view loading is not started soon enough or something along those lines.

**Comment by angusmccloud** [comment]
Subscribing to this chain as I'm seeing the same issue on iOS (and annoyingly don't have the same background color on the entire screen, so just formatting the background isn't a great fix).

**Comment by patrickkempff** [comment]
For an internal project we implemented a quickfix by sending a signal in the componentDidMount method of the Screen class and having the navigation controller listen for this before pushing the corresponding view controller. I believe airbnb's native navigation does something comparable. Although it feels hacky, I can share the solution if anybody is interested?

**Comment by dvorakjan** [comment]
@patrickkempff Sounds good, I am interested. We were forced to downgrade to non-native react-navigation because of this.

**Comment by angusmccloud** [comment]
@patrickkempff I'd love to see the hacky-way you're doing it! This issue isn't going to stop me from using this navigator, but some of our users have been pointing it out.

Thanks!

**Comment by jbolter** [comment]
@patrickkempff I am also definitely interested in your "hacky" solution to this.  I've been meaning to look into this but haven't had a time to create a hack of my own.

**Comment by patrickkempff** [comment]
Cool! Busy week. I will add the code to github this weekend.

**Comment by jbolter** [comment]
@patrickkempff Any chance you could share that workaround "hack" with us? Thanks!

**Comment by decodez** [comment]
@patrickkempff It would be a great help if you could share that workaround soon with us. Much appreciated!!

**Comment by melihberberolu** [comment]
@patrickkempff can u share your hacky solution ?

**Comment by melihberberolu** [comment]
What is the tricky way? I'm used interaction manager and backgroundColor but nothing change.

**Comment by steven-luu** [comment]
I'm confused about something in this thread, so I'd like to confirm..  

`screenBackgroundColor` was added to navigatorStyle and it certainly solves the problem of "white flash on screen push".  It gets applied to the UIView on the `viewWillAppear` event, so this should work as intended.

We're now using 1.1.178, and we don't see this issue.  Are you guys talking about v2 ?

**Comment by angusmccloud** [comment]
@steven-luu I see that as a band-aid. Instead of seeing a white-flash you see a [customized-color]-flash. My app had the top 1/4-1/3 of the page a different color than the rest of the page (the header with a title, image, and search-box with content below) - so then I can set it to either show a white-flash over the header or a colored-flash over the lower body, but not neither (like on Android where there's no flash.

**Comment by grundmanise** [comment]
@angusmccloud  https://wix.github.io/react-native-navigation/#/styling-the-navigator?id=style-object-format -> `screenBackgroundImageName`

**Comment by angusmccloud** [comment]
I'll take a look, thanks @grundmanise

**Comment by jbolter** [comment]
Is there no solution to the actual "flash" other than to hide it with a solid color?  With true native iOS development I never see a flash when pushing view controllers.  Is this because the react JS engine has to render the screen while the push animation is happening?

**Comment by jslz** [comment]
Feel like I am seeing similar thing: every pushed screen is white until after the slide in animation, and then it renders? Would it somehow be possible to run the render step ahead of time? Is there some locking / single threaded ness that prevents them / makes it feel too delayed? Thanks for any thoughts. Will be trying the solid color at least for now.

**Comment by fokoz** [comment]
I have tried the componentDidMount hack as well, sadly it doesn't 100% work.
Flicker still happens about 20% of times. I feel like it is not about whether react view loaded or not.

Setting up delay before pushing (150ms+) is almost 100% work but users may feel it.

I love this package and still have a faith for the chosen one who will figure this out.

**Comment by cvarley100** [comment]
@fokoz @patrickkempff What method were you using with componentDidMount?

My app uses an image of a gradient as its background. I'm currently using a combination of the navigator style props screenBackgroundColor and screenBackground, as well as setting the ImageBackground backgroundColor to the dominant gradient colour. 

The white flash is gone, but there's a noticeable transition between the single-colour backgroundColor and the gradient image.

**Comment by fokoz** [comment]
@cvarley100
I put some codes into RCCNavigationController by using queue, when I call the push code of RNN(`this.props.navigator.push`), I hold the transition by queue it in native code and on `componentDidmount` of the pushed element, I then send some signal to native code to start the normal transition.

Without `setTimeout` on `componentDidmount`, the flicker happens about 20-30%, with 100ms `setTimeout`  the percentage has dropped to 15-20%.

One more thing about using `screenBackgroundColor`, be aware of this issue https://github.com/wix/react-native-navigation/issues/1047

**Comment by MrLoh** [comment]
@patrickkempff please share your solution, just copy paste the native code you wrote somewhere, it doesn't matter wether it's nice or anything, but it would be so helpful. 

This issue is quite annoying and a hackt fix is certainly much better than the current situations. My users won't care about the code quality, I promise.

**Comment by jaygould** [comment]
Not sure if this will help anyone, but Wix have also made rn-synchronous-render which claims to help with this problem: [https://github.com/wix/rn-synchronous-render](https://github.com/wix/rn-synchronous-render) .

I've been looking for a solution to this too but not yet tried this one.

**Comment by DanielZlotin** [comment]
Yes that was the main idea behind the synchronous render proof of concept. It will probably not solve the problem completely though, as current investigation shows that the render time itself (in js thread) is very slow. Further research is needed.

**Comment by mobitar** [comment]
A workaround for me has been to divide the rendering of my component into different times. First render the easy stuff, like some static cells/headers, then perform the rest of the render after a timeout.

Something like:

```
constructor(props) {
    super(props);
    var staticItems = [1, 2, 3];
    this.state = {staticItems: staticItems}
}

componentDidMount() {
  setTimeout(function () {
    this.setState(functionm(prevState){
       return _.merge(prevState, {dynamicItems: Server.getData()})
    })
  }.bind(this), 10);
}
```

RN will render the static stuff really quickly; quick enough that you won't see the white flash anymore. After that, you can load in the remaining data.

**Comment by Soraph** [comment]
I hope is not too late, but I managed to fix this in iOS by abusing of `screenBackgroundImageName` and `rootBackgroundImageName`.

I had the white flash when transitioning to a screen with these styles:
```
```

**Comment by MPiccinato** [comment]
Here is a temporary fix I have put in place after I copied the repo into our own codebase for the time being. The basic idea is to wait for the React Native view in iOS to finish rendering before pushing it on the navigation stack.

This can however cause a performance issue, a long delay on the push, if you are doing too much work in your components render method. A strategy we implemented on our main views for controllers was to set a timeout of 100ms before doing any heavy lifting. Another strategy was to properly guard from redux updates :)

https://gist.github.com/MPiccinato/7436301a8292af653dfdd628ffaa3139

Line 16: New bool `_rendering` to signal a controller being pushed is rendering

Line 67: Listens to RCTContentDidAppearNotification and calls `contentDidAppear:`

Line 72: `contentDidAppear` dispatches a function on the main queue and sets `_rendering` to NO

Line 184: Set `_rendering` to YES prior to pushing the controller

Line 442: Add `_rendering` to conditional to see if the controller should be queued instead of pushed immediately

Line 451: Set `_rendering` to NO to allow for the next controller to push

**Comment by cvarley100** [comment]
@MPiccinato Thank you for this.

The only issue is that on some screens, there is a flash of the page content in the top left hand corner on navigation push.

Do you know why that might be?

**Comment by alien3d** [comment]
thanks @MPiccinato .. But seem a little heavy.. not sure i will try on my phone.

**Comment by MPiccinato** [comment]
I have updated the gist to allow for only one VC to be pushed/queued at a time.

@cvarley100 Sorry, I haven't experienced that issue :( The only thing I can think of is some sort of screen sizing, height/width issue in the styles?

@alien3d It is a bit heavy but got us to a dependable working state. Since the library is going through a rewrite we have pulled all of the current branches code into our repo.

**Comment by alien3d** [comment]
@MPiccinato  are you mean v2  react native navigation.. I yet not migrate to v2 version.. I have test to my mobile .Pretty slow not usable .. I'm not sure because i redirect to the same screen since upon push to other screen no weird white screen.

**Comment by hellomaya** [comment]
I have seen the same issue here with 1.1.256, I am using one of the oldest available iOS devices (ipod touch 5) to run the app in release, I saw the code to run the JS component when the RCTContentView added, as JSC is a single thread engine, it will try do "runApplication" (JS) one by one, that means, the last JS View in last tab will be loaded last.

As the device is quite old, so when I open the app, the first JS View in first Tab is there, just like what React Native do, one RCTRootView, no problem. But if I quickly switch to the last one, then it will show flash before the JS View loaded.

And even more, if I push a JS View with React Navigation Bar, then it will show white page, and then the JS View showed after about 150ms, still white flash.

The second issue is that, the Navigation Controller will initialize RCTRootView at run-time, so the "runApplication" JS will be called, but not wait for its complete, then push it to the stack, that's even worse, because Native navigation is much more quicker than JS based navigation, so flashing again.

This happened in JS based Navigation framework too, if the JS View is heavy, the navigation transition will be obviously laggy, because the JS View have no enough time to mount itself yet.

One solution in JS based navigation is to use "delay" before push it, also use "light" JS View, to keep transition smooth.

In React native navigation, it also worked by using "delay" in my device and app, in RCCNavigationController, I keep it wait for 150ms, then load it, it was good.

```
 dispatch_time_t time = dispatch_time ( DISPATCH_TIME_NOW , 150ull * NSEC_PER_MSEC) ;
      
      dispatch_after ( time , dispatch_get_main_queue ( ) , ^ {
        
        [self pushViewController:viewController animated:animated];
        
      } ) ;
```

Unfortunately I read code and never found any simple way to fix tab switch flashing issue, I also found that seems, it's not possible to hook into JS based component, and RCTContentView still use bridge to run JS View and it's in another thread, so there is no way to know the JS component is loaded without changing React Native. 

By using screenBackgroundColor will make the app blank sometimes too.

**Comment by MPiccinato** [comment]
@alien3d This is for V1

**Comment by hellomaya** [comment]
I will share my temp fix, well you can try it but but to be caution, this fix will have to add something new in React library.

- The solution is based on @guyca idea, to check the React Component is added, and then go push action

To do it in iOS:

- Adding a Category for RCTRootContentView in React/Base, as it's the content View of React Root View
```
#import "RCTRootContentView+RCTRootContentViewExtension.h"

@implementation RCTRootContentView (RCTRootContentViewExtension)

/// @@@Tom
#if true
- (void)didAddSubview:(UIView *)subview
{
  [[NSNotificationCenter defaultCenter]
   postNotificationName:@"DidAddReactComponent"
   object:self userInfo:@{@"view": subview}];
}
#endif

@end
```
- Added method didAddSubview, this method will be called when there is a subview added to content view, so if a React Component is added, it will be called many times (usually the component had many views)

Then, go react native navigation 

- in RCCNavigationController.m performAction method:

```
RCCViewController *viewController = [[RCCViewController alloc] initWithComponent:component passProps:passProps navigatorStyle:navigatorStyle globalProps:nil bridge:bridge];
    viewController.controllerId = passProps[@"screenInstanceID"];
    
    
    /// @@@ Tom
    _action = [performAction copy];
    _rccViewController = viewController;
    /// @@@

```

Added two private member to keep the viewController and action in context.

- Same file, in initWithProps added to observe the Notification:

```
 [[NSNotificationCenter defaultCenter] addObserver:self
                                           selector:@selector(didAddReactComponent)
                                               name:@"DidAddReactComponent"
                                             object:nil];
```

- Same file, added a method to deal with push when component is added once

```
- (void)didAddReactComponent
{
  // RCTAssertMainQueue();
  // Clear the reactTag so it can be re-assigned
  // self.reactTag = nil;
  // NSLog(@"Hi, there");
  
  
  if ([_action isEqualToString:@"push"]) {
    
    if (_rccViewController == nil) {
      return;
    }
    
    // NSLog(@"Hi, its push!!");
    [self pushViewController:_rccViewController animated:TRUE];
    _action = @"";
  }
}
```

Keep in mind, have to deal with this one time only.

In very old iOS device, touch Tab, or show Modal all showing blinks, Tab usually better, because it run the JS based component at initial time, but modal have same problem, I didn't find how to fix modal yet.

Hope it helps.

**Comment by DevGentle** [comment]
@MPiccinato I try to do the same code as your `RCCNavigationController.m` and nothing happen can you recommend me what I need to do now ?

**My step**
1. open file `RCCNavigationController.m` from `node_modules/react-native-navigation/ios/RCCNavigationController.m`
2. copy your code in file `RCCNavigationController.m`
3. paste on entire of file `RCCNavigationController.m`

That's all of my step.

**Comment by ArtiomShapovalov** [comment]
I'm still experiencing the same issue. Will it be fixed in future?

**Comment by camdagr8** [comment]
Any movement on this?

**Comment by nezaidu** [comment]
Anything?

**Comment by MPiccinato** [comment]
@nezaidu I don't think there is much dev on v1 happening, I believe v2 is being focused on. You can try my solution here, https://github.com/wix/react-native-navigation/issues/358#issuecomment-337598200

**Comment by nezaidu** [comment]
@MPiccinato thanks!

**Comment by acollazomayer** [comment]
@MPiccinato i try your solution and it did not work. The application seems laggie, i have to press the push screen button two times to change screen, and the white flicker is still there. Anyways thank you

**Comment by shahen94** [comment]
Any fixes here?

**Comment by EskelCz** [comment]
For V2 the fix is
```
  Navigation.setDefaultOptions({
    layout: {
      backgroundColor: 'black'
    }
  })
```

**Comment by ryanscottaudio** [comment]
^ is not a fix, as it just creates a black flash instead of a white one.

**Comment by LRNZ09** [comment]
Still having this issue when using an image 😞

**Comment by xvonabur** [comment]
@MPiccinato Thanks! There is one problem with your approach and `setButtons`, because it sets buttons on visible screen, not on the new one. I can add `setTimeout` for it, but it's not ideal solution. Is there a way to postpone `setButtons` for corresponding screen until it renders?

**Comment by VicFrolov** [comment]
Any potential updates/solutions for V2. Setting a backgroundColor doesn't particularly help, because if the backgroundColor is already white, the segue occurs, no content, and then content pops in.

**Comment by git-pod2** [comment]
navigatorStyle: { navBarHidden : true, screenBackgroundColor: '#000000' } worked for me. However, navigatorStyle: { navBarHidden : true, screenBackgroundColor: 'black' } did NOT. On v1

**Comment by LRNZ09** [comment]
This is still relevant.

**Comment by VicFrolov** [comment]
Relevant indeed,

**Comment by rvkuzmik** [comment]
Still a problem.

**Comment by zazagag** [comment]
Guys, any news? Still a problem.

**Comment by 8BallBomBom** [comment]
Still a problem with the latest release.
Would be nice to see a fix for this.

**Comment by pascalgagneur** [comment]
Still having this issue on react-native 0.57.3 with react-native-navigation 1.1.493

**Comment by ryanscottaudio** [comment]
@guyca unless i'm misunderstanding #3661, this is a totally different issue; #3661 seems to be centered on initial loading of the app, while this issue is centered on the pushing of any new screen.

**Comment by loganthompson** [comment]
> navigatorStyle: { navBarHidden : true, screenBackgroundColor: '#000000' } worked for me. However, navigatorStyle: { navBarHidden : true, screenBackgroundColor: 'black' } did NOT. On v1

navigatorStyle is definitely your fix on v1. This works great either on push or in screen setup.

**Comment by henrikra** [comment]
Does anyone have some idea where the code is that needs to be changed? I can look into that because I am facing the same problem. I would just want to be shown to right direction :)

**Comment by ryanscottaudio** [comment]
@loganthompson what happens when my screen is half one color, half another? what happens when it's a gradient? what happens when it's an image, or buttons, or really anything besides just a solid color? having only a solid color as an option isn't a good solution IMO; screens in native apps are too complex for that.

**Comment by pascalgagneur** [comment]
Our problem where related to https://github.com/facebook/react-native/issues/22259
By switching webview to https://github.com/react-native-community/react-native-webview fixed out problem.

**Comment by rachydro** [comment]
does anyone have solutions for V3?

**Comment by jmacioszek** [comment]
@guyca I agree with @ryanscottaudio - this is a differnet issue and still relevant

**Comment by jmacioszek** [comment]
@ryanscottaudio this can be solved by a secret not documented on push option:

`
Navigation.push(_, { 
  component: {
     ...,
     options: {
        animations: {
           push: {
              waitForRender: true
           }
        }
     }
  }
})
`

It may be cause delays in pushing new screen on slower devices though

**Comment by stokesbga** [comment]
@jmacioszek how the fuck? Nice find man

**Comment by larryranches** [comment]
> @ryanscottaudio this can be solved by a secret not documented on push option:
> 
> ```
> Navigation.push(_, { 
>   component: {
>      ...,
>      options: {
>         animations: {
>            push: {
>               waitForRender: true
>            }
>         }
>      }
>   }
> })
> ```
> It may cause delays in pushing new screen on slower devices though

This solution does seems to work for `showModal` animation `waitForRender: true` property and you can see that is working when setting the boolean from false to true. 

What I found is that only partially works on the `push animation waitForRender: true` at least for `v2.13.0`. When setting this on push, there is still a white flicker but the flicker seems to happen at a faster speed but is still very noticeable when setting the boolean from false to true.

Any ideas?

**Comment by davidmarinangeli** [comment]
As for the `showModal` problem:[ we solved removing two lines ](https://github.com/wix/react-native-navigation/issues/5262#issuecomment-525710268) from `AppDelegate.m`. I don't know if it could be useful to somebody

**Comment by juancarlosgzrz** [comment]
I'm seeing this when I switch to another page.  I don't know what React native is, but I'm seeing this. Is there a way to fix it?

**Comment by nihp** [comment]
Any solution to not show the splash in react-navigation version 3

**Comment by acollazomayer** [comment]
@nihp swith to react-navigation. Far better API and now works smoothly. Switch is not that hard

**Comment by glintpursuit** [comment]
Any solution for this ? 
We are facing same issue & above mention solution does not work for us. 
 "react-native": "0.61.5",
"react-navigation": "^4.0.10",

**Comment by msqar** [comment]
It happens to me too, using 3.7.0 react-native-navigation and RN 0.60.6, whenever i push to new screen, it shows blank until it renders the content. It's annoying! Attempted setting the waitForRender in true for every single option within stack but not working at all. Any ideas?

**Comment by nihp** [comment]
We also facing the same issue

**Comment by guyca** [comment]
@nihp You can use `waitForRender` [option](https://github.com/wix/react-native-navigation/issues/1086#issuecomment-409995258)

**Comment by nihp** [comment]
@guyca I have tried waitForRender in my app. It does not work for me

  animations: {
        push: {
          waitForRender: true
        },
        showModal: {
          waitForRender: true
        }
      },

**Comment by glintpursuit** [comment]
@nihp 

https://facebook.github.io/react-native/docs/running-on-device#pro-tip

// Place this code after "[self.window makeKeyAndVisible]" and before "return YES;"
UIView* launchScreenView = [[[NSBundle mainBundle] loadNibNamed:@"LaunchScreen" owner:self options:nil] objectAtIndex:0];
launchScreenView.frame = self.window.bounds;
rootView.loadingView = launchScreenView;

**Comment by msqar** [comment]
@nihp have you tried putting the waitForRender: true whenever you are about to push to a new screen? It worked for me that way, it was the only way i could make it work. Whenever I do a Navigation.push() i set the waitForRender within the options. Works like a charm. Good luck!

**Comment by nihp** [comment]
@msqar 
I have set in my routes as below mentioned code.

Also, I have tried with the each createStackNavigator ==> navigation options too.

  ```
{
    initialRouteName: 'AuthScreen',
    headerMode: 'none',
    waitForRender: true
  },
```

**Comment by nenjamin2405** [comment]
I'm using v5.1.1. I've tried the solution
```
Navigation.push(_, { 
  component: {
     ...,
     options: {
        animations: {
           push: {
              waitForRender: true
           }
        }
     }
  }
})
```

But it didn't work. Any idea? :(

**Comment by AdarshJais** [comment]
> Hey guys this issue will be addressed soon, I hope by the end of November. Sorry for the inconvenience.
> 
> @savelichalex you're correct, that's exactly what happens. On Android we avoid this issue by showing the pushed screen only after `render` is done and the native view is populated.

may I know how you did so in Android?

**Comment by AdarshJais** [comment]
> I'm using v5.1.1. I've tried the solution
> 
> ```
> Navigation.push(_, { 
>   component: {
>      ...,
>      options: {
>         animations: {
>            push: {
>               waitForRender: true
>            }
>         }
>      }
>   }
> })
> ```
> 
> But it didn't work. Any idea? :(

Did you got the solution ??

**Comment by jmhuret** [comment]
I've found that using the `waitForRender: true` can be useful, but also just setting the layout's componentBackgroundColor also works. I'm using RNN 6.12.2 for reference.

```
options: {
  layout: {
    componentBackgroundColor: Theme.colors.background,
  },
},
```

**Comment by worstpractice** [comment]
In the `screenOptions` prop passed to `Stack.Navigator`, setting `stackAnimation: "none"` fixes the white flashes for me.

Sure, the resulting navigations do become jarringly instant - but I've never heard a user complain that an app is "too snappy" or "responds too quickly" to interactions.

I'm using Expo 39.0.4 with `@react-navigation/native` version `5.8.10`.

**Comment by chrise86** [comment]
> In the `screenOptions` prop passed to `Stack.Navigator`, setting `stackAnimation: "none"` fixes the white flashes for me, atleast on Android (physical device). I've yet to examine what effect this tweak might have on other platforms.
> 
> 
> 
> Sure, the resulting navigations do become jarringly instant - but I've never heard a user complain that an app is "too snappy" or "responds too quickly" to interactions.
> 
> 
> 
> I'm using Expo 39.0.4 with `@react-navigation/native` version `5.8.10`.

@worstpractice that is a different library, not this one.

**Comment by Abhishek-Sankey** [comment]
> @ryanscottaudio this can be solved by a secret not documented on push option:
> 
> ```
> Navigation.push(_, { 
>   component: {
>      ...,
>      options: {
>         animations: {
>            push: {
>               waitForRender: true
>            }
>         }
>      }
>   }
> })
> ```
> 
> It may cause delays in pushing new screen on slower devices though

maybe I'm asking a silly question but where should I write this code in the project?

**Comment by chrise86** [comment]
@Abhishek-Sankey I believe you'll need to use that snippet (everything inside the `options` object) whenever you push a screen that suffers from the white flash and/or you want to wait for the render to complete before the push/navigation.
